reactive/containerd.py

```diff
@@ -103,14 +105,40 @@ def charm_status():
         status.blocked('Container runtime not available')
 
 
-def merge_custom_registries(custom_registries):
+def update_custom_tls_config(config_directory, registries):
+    """
+    Read registries config and write tls files to disk.
+
+    :param str config_directory: containerd config directory
+    :param List registries: juju config for custom registries
+    :return: None
+    """
+    for registry in registries:
+        for opt in ['ca', 'key', 'cert']:
+            file_b64 = registry.get('%s_file' % opt)
+            if file_b64:
+                try:
+                    file_contents = base64.b64decode(file_b64)
+                except (binascii.Error, TypeError):
+                    continue
```

Comment:
if it isn't base64, just ignore it and look into the next thing
would a warning in the debug log useful here maybe?

Reply:
Logging a warning is ok, but it would be nice if this could be surfaced as a `blocked` status message, since it won't be obvious to the operator what went wrong and might take some time tracking it back to the log message.